src/panels/lovelace/cards/hui-grid-card.ts

```diff
@@ -0,0 +1,65 @@
+import { css, CSSResult } from "lit-element";
+import { computeCardSize } from "../common/compute-card-size";
+import { HuiStackCard } from "./hui-stack-card";
+import { GridCardConfig } from "./types";
+
+const DEFAULT_COLUMNS = 3;
+
+class HuiGridCard extends HuiStackCard<GridCardConfig> {
+  public async getCardSize(): Promise<number> {
+    if (!this._cards || !this._config) {
+      return 0;
+    }
+
+    const promises: Array<Promise<number> | number> = [];
+
+    for (const element of this._cards) {
+      promises.push(computeCardSize(element));
+    }
+
+    const results = await Promise.all(promises);
+
+    const maxCardSize = Math.max(...results);
+
+    return (
+      maxCardSize *
+      (this._cards.length / (this._config.columns || DEFAULT_COLUMNS))
```

Comment:
Should we limit the columns to the number of cards when there are less cards than columns?
```suggestion
      (this._cards.length / (this._config.columns || (Math.min(DEFAULT_COLUMNS, this._cards.length)))
```

Reply:
^ thats not going it work reading the code below but still... 😛

Reply:
So if we want that, we can't expose the CSS var. I'm fine either way. What do you prefer?

Reply:
I think it is nice for a user to not have to worry about the columns setting, and if you add 2 cards you don't get a blank space. But also to make it work like our current horizontal stack.

Reply:
I think that it depends on if you want this. For example, if I have 2 grid cards next to one another, I want all cards to be the same size!

<img width="1600" alt="image" src="https://user-images.githubusercontent.com/1444314/97415004-fa030e80-1904-11eb-9cb0-cf5f8934a876.png">

Reply:
Oh I guess we already don't have that now
<img width="1064" alt="image" src="https://user-images.githubusercontent.com/1444314/97415257-45b5b800-1905-11eb-8e23-2460024d7dd2.png">

Reply:
Back to the drawing board. I think that we should make sure that the height is included (but removable maybe?) to have multiple grid cards look the same.

Reply:
Better: 
<img width="1019" alt="image" src="https://user-images.githubusercontent.com/1444314/97416655-ff615880-1906-11eb-9edf-3af87105bc7f.png">